app/routes.py

```diff
@@ -1,2 +1,294 @@
+            db.session.add(new_task)
+            db.session.commit()
+
+            # return jsonify(task= {
+            # "id": new_task.id,
+            # "title": new_task.title,
+            # "description": new_task.description,
+            # "is_complete": True if new_task.completed_at is not None else False
+            # }), 201
+            return make_response({"task" : new_task.to_dict()}, 201)
+
+
+'''
+Get One Task, Update One Task, Delete One Task
+'''
+@tasks_bp.route("/<task_id>", methods=["GET","PUT","DELETE"], strict_slashes=False)
+def handle_single_task(task_id):
+
+    # Find the task with the given id
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return make_response("", 404)
+        # return make_response(jsonify(None), 404)
+
+    if request.method == "GET": 
+        # return make_response(jsonify(task= {
+        # "id": task.id,
+        # "title": task.title,
+        # "description": task.description,
+        # "is_complete": True if task.completed_at is not None else False
+        # }), 200)
+        return {"task" : task.to_dict()}
+    
+    elif request.method == "PUT":
+        new_data = request.get_json()
+
+        task.title = new_data["title"]
+        task.description = new_data["description"]
+        task.completed_at = new_data["completed_at"]
+
+        db.session.commit()
+
+        return make_response({"task" : task.to_dict()}, 200)
+
+        # return jsonify(task= {
+        #     "id": task.id,
+        #     "title": task.title,
+        #     "description": task.description,
+        #     "is_complete": True if task.completed_at is not None else False
+        # }), 200
+
+        # return make_response(jsonify(task=task.to_dict()), 200)
+        
+    elif request.method == 'DELETE':
+        db.session.delete(task)
+        db.session.commit()
+
+        return make_response({"details": f"Task {task.id} \"{task.title}\" successfully deleted"}, 200)
+        # return make_response(jsonify(details=f"Task {task.id} \"{task.title}\" successfully deleted"), 200)
+
+'''
+Wave 03: Mark a task complete or incomplete
+'''
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"], strict_slashes=False)
+def mark_complete(task_id):
+
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return make_response("", 404)
+
+    if request.method == "PATCH":
+
+        task.completed_at = datetime.datetime.now()
+
+        db.session.commit()
+
+        send_slack(task)
+
+        return {"task" : task.to_dict()}
+
+        # return jsonify(task= {
+        #     "id": task.id,
+        #     "title": task.title,
+        #     "description": task.description,
+        #     "is_complete": bool(task.completed_at)
+        # }), 200
+    
+    return("PATCH method only. Do not accept other methods for this route.")
+
+'''
+Wave 04: Notify user a task was completed with Slack messages, using the Slack web API.
+'''
+def send_slack(task):
```

Comment:
Love that this is its own function! I think that makes the rest of the code easier to read :)